### ISSUE TITLE
Disabling column auto-resize

### ISSUE THREAD

**Issue by nartz** [arx-deidentifier/arx#278]
**Problem**
We are importing data where some columns contain long texts (like "comments")Upon importing the data, ARX resizes the columns in the main Data Table fit the long texts in cells, and columns are very long and make viewing data difficult, requiring lots of side scrolling.

**Describe the solution you'd like**
We'd like to turn off auto-resizing when importing data, instead defaulting to setting a max column width and showing abbreviated elipses or truncation when columns are smaller than total content contained. Columns should still be resizable by manually making them bigger / smaller.

**Describe alternatives you've considered**
We notice that ComponentTable used for Hierarchy seems to implement what we want (truncation / abbreviation using elipses), however, ComponentDataTable (we believe used for the main input data viewer) doesn't? We've looked at the code some but haven't pointpointed what to change. Something with the styling or configuration of the nattables we believe (AutoResizeColumnAction?), but aren't quite sure which to look at!

**Additional context**
We are not sure if this is a real feature request that the larger community needs, or just something we would like. Just any suggestion about what (few?) lines to look at might be helpful!

**Comment by prasser** [comment]
Thanks for reporting this! This is definitely an enhancement that would be relevant for all users.

ComponentDataTable and the classes in the package gui.view.impl.commin.datatable are the right place to look at. I don't have any more concrete pointers, however, as NatTable is a bit tricky to configure and we wrote this code years ago.

A PR would be great!

**Comment by nartz** [comment]
https://www.eclipse.org/nattable/documentation.php?page=faq has a section on "How to autoresize columns/rows programmatically?"

So, it appears if we just comment out ComponentDataTable.java lines 367/371
`
/       natTable.getConfigRegistry().registerConfigAttribute(CellConfigAttributes.CELL_PAINTER,
                                                             new DataTableDecorator( new TextPainter(false, true, 0, true),
                                                             new BorderStyle(2, GUIHelper.COLOR_BLACK, LineStyleEnum.SOLID)),
                                                                DisplayMode.NORMAL,
                                                                GridRegion.BODY);
`

Then it seems to behave as requested. However, i'm not fully sure of the ramifications of this (aside from the positive side effects we want)

**Comment by nartz** [comment]
Ah stupidly - so there is a boolean flag on the TextPainter constructor, (the 4th arg), which is whether or not it should calculate size. Turning this to false disables the auto-size calculation.

new TextPainter(false, true, 0, true) -> new TextPainter(false, true, 0, **false**)

**Comment by nartz** [comment]
#279

**Comment by prasser** [comment]
Thanks. I didn't take a look at the result, but I would like to point out that the optimal solution would probably be to still calculate the width of columns but to introduce a maximum size. This could probably be implemented by overriding getPreferredWidth() in the TextPainter. See https://github.com/eclipse/nebula.widgets.nattable/blob/master/org.eclipse.nebula.widgets.nattable.core/src/org/eclipse/nebula/widgets/nattable/painter/cell/TextPainter.java

Have you tried that?

**Comment by nartz** [comment]
Haven't tried that at all - for our use case this is enough at this point.
This approach is also simpler and has the added benefit of loading faster,
since it doesn't need to calculate cell sizes at all - i think this is also
what excel does by default. It still supports double clicking on a column
to resize. However, I agree with you theoretically on your approach, but am
having a hard time finding a use case where it would actually provide much
value - but I could be wrong.

As a short thought example, if the columns have long text, I'm not going to
be able to read them all anyway if they are truncated, so the approaches
are about equal.
Similarly, if the columns have shorter texts, well, there is a little extra
padding which theoretically could be prune down, but I notice visually it's
easier to scan when all the column widths are equal (as in excel). If those
columns containing short texts were resized to be smaller, in support of
your argument, the user *might* be able to see another half column or full
column in the field of view - but it just depends on their dataset.

Anyway, just a few thoughts.

**Comment by prasser** [comment]
Fixed.

**Comment by prasser** [comment]
I reopened this issue, as I just realized that there is another issue. Your issue referred to the values within the containing long texts, right? This is fixed by your PR. However, ARX doesn't allow making the width of columns so small that the text in the column header cannot be displayed fully. This will lead to problems with long texts in columns. Did you encounter this problem before?

**Comment by nartz** [comment]
Let me see if I understand properly. The issue is that the width of the
columns cannot currently be smaller than the width of the table headers,
correct? I'm not sure i'm following how this will lead to problems with
long texts though - could you clarify the scenario you are thinking of?

**Comment by prasser** [comment]
Yes. Think about a column with a very long header. The column will have a very large width and it will not be possible to make the column any smaller.

**Comment by prasser** [comment]
BTW: This is unrelated to your change, just another, very similar, problem.

**Comment by nartz** [comment]
I see - yes, i could see that being an issue and agree. Might be able to
take a look into that soon.

**Comment by prasser** [comment]
@nartz Did you come up with a solution?

**Comment by nartz** [comment]
Looked into it briefly but don't have a fix yet.
